src/components/Admin/Categories.js

```diff
@@ -32,17 +26,52 @@ class Categories extends Component {
   }
 
   handleOk = (e) => {
+    this.props.addCategory(this.state.newCategoryName);
     this.setState({
       visible: false,
+      newCategoryName: ''
     });
   }
 
   handleCancel = (e) => {
     this.setState({
       visible: false,
+      newCategoryName: ''
+    });
+  }
+
+  updateShowModal = () => {
+    this.setState({
+      visibleUpdateModal: true,
+    });
+  }
+
+  handleOkUpdate = (id) => {
+    this.props.updateCategory(this.state.updateCategoryName,id);
+    this.setState({
+      visibleUpdateModal: false,
+      newCategoryName: ''
+    });
+  }
+
+  handleCancelUpdate = (e) => {
+    this.setState({
+      visibleUpdateModal: false,
+      newCategoryName: ''
     });
   }
 
+  handleDelete = (id) => {
+    this.props.deleteCategory(id);
+  }
+
+  componentWillMount(){
+    this.props.fetchCategory();
+  }
+  componentDidUpdate(){
```

Comment:
No need to call the fetch categories in the didUpdate lifecycle , because it is already present in the state when fetched from willMount

Reply:
Done